cns/restserver/util.go

```diff
@@ -201,7 +206,33 @@ func (service *HTTPRestService) getNetworkContainerResponse(req cns.GetNetworkCo
 		}
 
 		logger.Printf("pod info %+v", podInfo)
-		containerID = service.state.ContainerIDByOrchestratorContext[podInfo.PodName+podInfo.PodNamespace]
+
+		context := podInfo.PodName + podInfo.PodNamespace
+		containerID, exists = service.state.ContainerIDByOrchestratorContext[context]
+		if service.ChannelMode == cns.Managed {
+			if exists {
+				_, getNetworkContainerResponse.Response.ReturnCode, getNetworkContainerResponse.Response.Message = isNCWaitingForUpdate(service.state.ContainerStatus[containerID].CreateNetworkContainerRequest.Version, containerID)
+				if getNetworkContainerResponse.Response.ReturnCode == Success {
+					return getNetworkContainerResponse
+				}
+			} else {
+				var (
+					dncEP     = service.GetOption(acn.OptPrivateEndpoint).(string)
+					infraVnet = service.GetOption(acn.OptInfrastructureNetwork).(string)
+					nodeID    = service.GetOption(acn.OptNodeID).(string)
```

Comment:
are we not using config?

Reply:
we do use config, those values are passed to the service